middleware/auth-utils/grant-manager.js

```diff
@@ -421,13 +421,19 @@ const fetch = (manager, handler, options, params) => {
 
     const req = getProtocol(options).request(options, (response) => {
       if (response.statusCode < 200 || response.statusCode > 299) {
-        return reject(response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ]);
+        let response = '';
+        response.on('data', (d) => (response += d.toString()));
+        response.on('end', () => {
+          const message = response.statusCode + ':' + http.STATUS_CODES[ response.statusCode ] + ' ' + response;
+          reject(new Error(message));
+        });
```

Comment:
IMO, this change will not work properly. you used variable `response` to store response's body and you  have response object with the same name.

Reply:
resolved